includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -504,47 +485,47 @@ public function should_show_payment_request_button() {
 		}
 
 		// Page not supported.
-		if ( ! $this->is_product() && ! $this->is_cart() && ! $this->is_checkout() ) {
+		if ( ! $this->express_checkout_helper->is_product() && ! $this->express_checkout_helper->is_cart() && ! $this->express_checkout_helper->is_checkout() ) {
 			return false;
 		}
 
 		// Product page, but not available in settings.
-		if ( $this->is_product() && ! $this->is_available_at( 'product' ) ) {
+		if ( $this->express_checkout_helper->is_product() && ! $this->express_checkout_helper->is_available_at( 'product', 'payment_request_button_locations' ) ) {
```

Comment:
Maybe we can put `payment_request_button_locations` to some class constant or variable to avoid typos. Same for `class-wc-payments-woopay-button-handler.php` and its `platform_checkout_button_locations`.

Reply:
Good idea! Added in [07e990b](https://github.com/Automattic/woocommerce-payments/pull/8055/commits/07e990bc5a5b7497f36832e044fb0884697253b6)